ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Ledger.hs

```diff
@@ -73,35 +78,71 @@ data HardForkLedgerError xs =
   | HardForkLedgerErrorWrongEra (MismatchEraInfo xs)
   deriving (Generic, Show, Eq, NoUnexpectedThunks)
 
+{-------------------------------------------------------------------------------
+  GetTip
+-------------------------------------------------------------------------------}
+
+instance CanHardFork xs => GetTip (LedgerState (HardForkBlock xs)) where
+  getTip = castPoint
+         . State.getTip (castPoint . getTip)
+         . hardForkLedgerStatePerEra
+
+instance CanHardFork xs => GetTip (Ticked (LedgerState (HardForkBlock xs))) where
+  getTip = castPoint
+         . State.getTip (castPoint . getTip . unComp)
+         . tickedHardForkLedgerStatePerEra
+
+{-------------------------------------------------------------------------------
+  Ticking
+-------------------------------------------------------------------------------}
+
+data instance Ticked (LedgerState (HardForkBlock xs)) =
+    TickedHardForkLedgerState {
+        tickedHardForkLedgerStateTransition :: !TransitionInfo
+      , tickedHardForkLedgerStatePerEra     ::
+          !(HardForkState_ LedgerState (Ticked :.: LedgerState) xs)
+      }
+  deriving (Generic)
+
+deriving anyclass instance
+     CanHardFork xs
+  => NoUnexpectedThunks (Ticked (LedgerState (HardForkBlock xs)))
+
 instance CanHardFork xs => IsLedger (LedgerState (HardForkBlock xs)) where
   type LedgerErr (LedgerState (HardForkBlock xs)) = HardForkLedgerError  xs
 
   applyChainTick cfg@HardForkLedgerConfig{..} slot (HardForkLedgerState st) =
-        fmap HardForkLedgerState
-      . State.sequence
-      . hczipWith proxySingle (tickOne ei slot) cfgs
-      . State.extendToSlot cfg slot
-      $ st
+      TickedHardForkLedgerState {
+          tickedHardForkLedgerStateTransition =
+            -- The use of the extended but unticked ledger state to derive the
```

Comment:
Comment: We are bundling a `TransitionInfo` with a ticked state, but _derive_ it from the unticked (though extended) state. That requires justification.